worlds/noita/Regions.py

```diff
@@ -1,48 +1,51 @@
 
+if TYPE_CHECKING:
+    from . import NoitaWorld
+else:
+    NoitaWorld = object
 
-def add_location(player: int, loc_name: str, id: int, region: Region) -> None:
-    location = Locations.NoitaLocation(player, loc_name, id, region)
-    region.locations.append(location)
 
-
-def add_locations(multiworld: MultiWorld, player: int, region: Region) -> None:
+def add_locations(world: NoitaWorld, region: Region) -> None:
     locations = Locations.location_region_mapping.get(region.name, {})
     for location_name, location_data in locations.items():
         location_type = location_data.ltype
         flag = location_data.flag
 
-        opt_orbs = multiworld.orbs_as_checks[player].value
-        opt_bosses = multiworld.bosses_as_checks[player].value
-        opt_paths = multiworld.path_option[player].value
-        opt_num_chests = multiworld.hidden_chests[player].value
-        opt_num_pedestals = multiworld.pedestal_checks[player].value
+        opt_orbs = world.options.orbs_as_checks.value
+        opt_bosses = world.options.bosses_as_checks.value
+        opt_paths = world.options.path_option.value
+        opt_num_chests = world.options.hidden_chests.value
+        opt_num_pedestals = world.options.pedestal_checks.value
 
         is_orb_allowed = location_type == "orb" and flag <= opt_orbs
         is_boss_allowed = location_type == "boss" and flag <= opt_bosses
+        amount = 0
         if flag == Locations.LocationFlag.none or is_orb_allowed or is_boss_allowed:
-            add_location(player, location_name, location_data.id, region)
+            amount = 1
         elif location_type == "chest" and flag <= opt_paths:
-            for i in range(opt_num_chests):
-                add_location(player, f"{location_name} {i+1}", location_data.id + i, region)
+            amount = opt_num_chests
         elif location_type == "pedestal" and flag <= opt_paths:
-            for i in range(opt_num_pedestals):
-                add_location(player, f"{location_name} {i+1}", location_data.id + i, region)
+            amount = opt_num_pedestals
+
+        region.add_locations(Locations.make_location_range(location_name, location_data.id, amount),
+                             Locations.NoitaLocation)
 
 
 # Creates a new Region with the locations found in `location_region_mapping` and adds them to the world.
-def create_region(multiworld: MultiWorld, player: int, region_name: str) -> Region:
-    new_region = Region(region_name, player, multiworld)
-    add_locations(multiworld, player, new_region)
+def create_region(world: NoitaWorld, region_name: str) -> Region:
+    new_region = Region(region_name, world.player, world.multiworld)
+    add_locations(world, new_region)
     return new_region
 
 
-def create_regions(multiworld: MultiWorld, player: int) -> Dict[str, Region]:
-    return {name: create_region(multiworld, player, name) for name in noita_regions}
+def create_regions(world: NoitaWorld) -> Dict[str, Region]:

```

Comment:
why does this function even exist?

Reply:
can't mark it, but same comment for create_item in Items.py line 25. that could just be the definition in the world and then all you have to do is `world.create_item(item_name)`

Reply:
> why does this function even exist?

To make `create_all_regions_and_connections` readable and keep the world class slim.